$NetBSD: patch-configure.ac,v 1.8 2022/07/27 19:53:16 wiz Exp $

Fix pkg-config'ing for lua.
Fix png library name.

--- configure.ac.orig	2022-04-18 22:10:15.000000000 +0000
+++ configure.ac
@@ -1401,7 +1401,7 @@ asm volatile("xorps %%xmm1,%%xmm2"::"r"(
     ])
   ])
 
-  AS_IF([test "${ac_cv_sse_inline}" != "no" -a "${SYS}" != "solaris"], [
+  AS_IF([test "${ac_cv_sse_inline}" != "no"], [
     AC_DEFINE(CAN_COMPILE_SSE, 1, [Define to 1 if SSE inline assembly is available.])
   ])
 
@@ -1416,7 +1416,7 @@ asm volatile("punpckhqdq %%xmm1,%%xmm2":
       ac_cv_sse2_inline=no
     ])
   ])
-  AS_IF([test "${ac_cv_sse2_inline}" != "no" -a "${SYS}" != "solaris"], [
+  AS_IF([test "${ac_cv_sse2_inline}" != "no"], [
     AC_DEFINE(CAN_COMPILE_SSE2, 1, [Define to 1 if SSE2 inline assembly is available.])
     have_sse2="yes"
   ])
@@ -1676,12 +1676,12 @@ AC_ARG_ENABLE(lua,
     [disable LUA scripting support (default enabled)])])
 if test "${enable_lua}" != "no"
 then
-  PKG_CHECK_MODULES(LUA, lua5.2,
+  PKG_CHECK_MODULES(LUA, lua-5.2,
     [ have_lua=yes ],
     [
     AC_MSG_WARN([${LUA_PKG_ERRORS}, trying lua 5.1 instead])
 
-    PKG_CHECK_MODULES(LUA, lua5.1,
+    PKG_CHECK_MODULES(LUA, lua-5.1,
       [ have_lua=yes ],
       [
       AC_MSG_WARN([${LUA_PKG_ERRORS}, trying lua >= 5.1 instead])
@@ -2843,7 +2843,7 @@ AS_IF([test "${enable_png}" != "no"], [
 AC_CHECK_HEADERS(png.h, [
   VLC_SAVE_FLAGS
   LDFLAGS="${LDFLAGS} -lz $LIBM"
-  AC_CHECK_LIB(png, png_set_alpha_mode, [
+  AC_CHECK_LIB(png16, png_set_alpha_mode, [
     VLC_ADD_PLUGIN([png])
   ], [], [-lz $LIBM])
   VLC_RESTORE_FLAGS
